htdocs/compta/facture/class/facture.class.php

```diff
@@ -2477,6 +2495,24 @@ public function validate($user, $force_number = '', $idwarehouse = 0, $notrigger
 				$this->setFinal($user);
                 }
 			}
+
```

Comment:
The code to generate PDF document is already called just outside the function validate on pages (if validate + triggers is success). So for mass action, it should be same and not be called inside the validate() by mass actions code (this will lead in duplicate PF generation in other cases.

Also, why did you disabled the
if ($conf->global->MAIN_MULTILANGS && empty($newlang)) ...